dot/core/service.go

```diff
@@ -548,3 +574,41 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	rt.SetContextStorage(ts)
 	return rt.Metadata()
 }
+
+// RegisterRuntimeUpdatedChannel function to register chan that is notified when runtime version changes
+func (s *Service) RegisterRuntimeUpdatedChannel(ch chan<- runtime.Version) (byte, error) {
+	s.runtimeUpdateSubscriptionsLock.Lock()
+	defer s.runtimeUpdateSubscriptionsLock.Unlock()
+
+	if len(s.runtimeUpdateSubscriptions) == 256 {
+		return 0, errors.New("channel limit reached")
+	}
+
+	var id byte
+	for {
```

Comment:
`generateID` should probably be a receiver function on the `Service` itself and you can check if there are collisions in the map so you don't have to loop here.

Reply:
Yes, that seems better.